book-content/chapters/03-typescript-in-the-development-pipeline.md

```diff
@@ -0,0 +1,244 @@
+
+### Initializing a TypeScript Project
+
+In order for TypeScript to know how to transpile our code, we need to create a `tsconfig.json` file in the root of our project.
+
+Running the following command will generate the `tsconfig.json` file:
+
+```bash
+tsc --init
+```
+
+Inside of the newly created `tsconfig.json` file, you will find a number of useful starter configuration options as well as many other commented out options.
+
+For now, we'll just stick with the defaults:
+
+```json
+// excerpt from tsconfig.json
+{
+  "compilerOptions": {
+    "target": "es2016",
+    "module": "commonjs",
+    "strict": true,
+    "esModuleInterop": true,
+    "skipLibCheck": true,
+    "forceConsistentCasingInFileNames": true
+  }
+}
+```
+
+With our `tsconfig.json` file in place, we can begin transpilation.
+
+### Running `tsc`
+
+Running the `tsc` command in the terminal without any arguments will take advantage of the defaults in `tsconfig.json` and transpile all TypeScript files in the project to JavaScript.
+
+```bash
+tsc
+```
+
+In this case, this means that our TypeScript code in `example.ts` file will become JavaScript code inside of `example.js`.
+
+Inside of `example.js`, we can see that the TypeScript syntax has been transpiled into JavaScript:
+
+```javascript
+// Inside of the example.js file
+
+"use strict";
+
+const run = (message) => {
+  console.log(message);
+};
+
+run("Hello!");
+```
+
+Now that we have our JavaScript file, we can update the `index.html` file to reference `example.js` instead of `example.ts`:
+
+```html
+// inside of index.html
+
+<script src="example.js"></script>
+```
+
+Opening the `index.html` file in the browser will now show the expected "Hello!" output in the console without any errors!
+
+### Does TypeScript Change My JavaScript?
+
+Looking at our JavaScript file, we can see how little has changed from the TypeScript code.
+
+```javascript
+"use strict";
+
+const run = (message) => {
+  console.log(message);
+};
+
+run("Hello!");
+```
+
+It's removed the `: string` from the `run` function, and added `"use strict";` to the top of the file. But other than that, the code is identical.
+
+This is a key guiding principle for TypeScript - it gives you a thin layer of syntax on top of JavaScript, but it doesn't change the way your code works. It doesn't add runtime validation. It doesn't attempt to optimise your code's performance.
+
+It just adds types (to give you better DX), and then removes them when it's time to turn your code into JavaScript.
+
+> There are some exceptions to this guiding principle, such as enums, namespaces and class parameter properties - but we'll cover those later.
+
+### A Note on Version Control
+
+We've successfully transpiled our TypeScript code to JavaScript, but we've also added a new file to our project. Adding a `.gitignore` file to the root of the project and including `*.js` will prevent the JavaScript files from being added to version control.
```

Comment:
This is true and all but wouldn't it be more useful to go straight with rootDir+outDir combination here?